Description: drop features depending on packages not in Debian yet
 Disable all providers except for mbed-crypto-provider. Namely:

 - pkcs11-provider
 - tpm-provider
 - trusted-service-provider
 - cryptoauthlib-provider

 In theory, all crates needed by trusted-service-provider are packaged in
 Debian. However, it needs files from another arm project called
 trusted-services-vendor, which has not been packaged yet.

 Also disable the jwt-svid authenticator.

 We will enable the missing features incrementally once the needed crates enter
 Debian.
Author: Emanuele Rocca <ema@debian.org>
Last-Update: 2023-06-19
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: parsec-service/Cargo.toml
===================================================================
--- parsec-service.orig/Cargo.toml
+++ parsec-service/Cargo.toml
@@ -29,10 +29,7 @@ repository = "https://github.com/paralla
 
 [package.metadata.docs.rs]
 features = [
-    "pkcs11-provider",
-    "tpm-provider",
     "mbed-crypto-provider",
-    "cryptoauthlib-provider",
     "direct-authenticator",
 ]
 
@@ -49,11 +46,6 @@ version = "0.13.0"
 [dependencies.bincode]
 version = "1.3.1"
 
-[dependencies.cryptoki]
-version = "0.3.1"
-features = ["psa-crypto-conversions"]
-optional = true
-
 [dependencies.derivative]
 version = "2.1.0"
 
@@ -77,18 +69,6 @@ version = "0.2.14"
 [dependencies.parsec-interface]
 version = "0.28.0"
 
-[dependencies.picky-asn1]
-version = "0.3.0"
-optional = true
-
-[dependencies.picky-asn1-der]
-version = "0.2.4"
-optional = true
-
-[dependencies.picky-asn1-x509]
-version = "0.6.1"
-optional = true
-
 [dependencies.prost]
 version = "0.11.0"
 optional = true
@@ -108,10 +88,6 @@ optional = true
 version = "0.28.0"
 #features = ["bundled"]
 
-[dependencies.rust-cryptoauthlib]
-version = "0.4.4"
-optional = true
-
 [dependencies.sd-notify]
 version = "0.4.0"
 
@@ -122,10 +98,6 @@ features = ["derive"]
 [dependencies.signal-hook]
 version = "0.3.4"
 
-[dependencies.spiffe]
-version = "0.2.1"
-optional = true
-
 [dependencies.structopt]
 version = "0.3.21"
 
@@ -135,10 +107,6 @@ version = "1.8.1"
 [dependencies.toml]
 version = "0.5.8"
 
-[dependencies.tss-esapi]
-version = "7.2.0"
-optional = true
-
 [dependencies.users]
 version = "0.11.0"
 
@@ -153,10 +121,6 @@ features = ["zeroize_derive"]
 version = "0.8.3"
 features = ["small_rng"]
 
-[dev-dependencies.rust-cryptoauthlib]
-version = "0.4.4"
-features = ["software-backend"]
-
 [build-dependencies.bindgen]
 version = "0.60.0"
 optional = true
@@ -169,39 +133,11 @@ optional = true
 all-authenticators = [
     "direct-authenticator",
     "unix-peer-credentials-authenticator",
-    "jwt-svid-authenticator",
 ]
 all-providers = [
-    "tpm-provider",
-    "pkcs11-provider",
     "mbed-crypto-provider",
-    "trusted-service-provider",
 ]
-cryptoauthlib-provider = ["rust-cryptoauthlib"]
-default = ["unix-peer-credentials-authenticator"]
+default = ["unix-peer-credentials-authenticator", "mbed-crypto-provider"]
 direct-authenticator = []
-jwt-svid-authenticator = ["spiffe"]
 mbed-crypto-provider = ["psa-crypto"]
-pkcs11-provider = [
-    "cryptoki",
-    "picky-asn1-der",
-    "picky-asn1",
-    "picky-asn1-x509",
-    "psa-crypto",
-    "rand",
-    "hex",
-]
-tpm-provider = [
-    "tss-esapi",
-    "picky-asn1-der",
-    "picky-asn1",
-    "picky-asn1-x509",
-    "hex",
-]
-trusted-service-provider = [
-    "psa-crypto",
-    "bindgen",
-    "prost-build",
-    "prost",
-]
 unix-peer-credentials-authenticator = []
